Main idea

xAceOfSpadesx16 opened the pull request:
Created:
- ColorModelField
- ColorChoicesWidget
- DropDown template
- Options template
- Abstract ColorModel

xAceOfSpadesx16 committed:
Main idea

Created:
- ColorModelField
- ColorChoicesWidget
- DropDown template
- Options template
- Abstract ColorModel

belthesar commented on colors/color_definitions.py:
ColorChoices assumes that Bootstrap5's colors are the only colors to choose from. Fine for MVP, but should be extended to add support for future CSS framework default colors, ideally with a setting configurable value to choose the framework to expose.

belthesar commented on pyproject.toml:
```diff
@@ -9,6 +9,9 @@ version = "2025.04.15"
 readme = "README.md"
 requires-python = ">=3.12"
+dependencies = [
+  "django",
```
django version should be more strict.

belthesar reviewed (changes requested):
Dang this is good work. Please see the comments for things I feel are important to change before acceptance.

xAceOfSpadesx16 committed:
added recomendations

added:
- default_color arg to ColorModelField (default BootstrapColorChoices)
- modified the way to get default options in formfield method.
- Especific framework TextChoices added as example for newer classes in case of framework compatibility

belthesar commented on pyproject.toml:
👍

belthesar reviewed (approved):
This is great Ace. Thank you!
